lib/web_ui/lib/src/engine/canvaskit/native_memory.dart

```diff
@@ -0,0 +1,206 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../engine.dart' show Instrumentation;
+import '../safe_browser_api.dart';
+import '../util.dart';
+import 'canvaskit_api.dart';
+
+/// Collects native objects that weren't explicitly disposed of using
+/// [UniqueRef.dispose] or [CountedRef.unref].
+SkObjectFinalizationRegistry _finalizationRegistry = SkObjectFinalizationRegistry(
+  allowInterop((UniqueRef<Object> uniq) {
+    uniq.collect();
+  }
+));
+
+/// Manages the lifecycle of a C++ object referenced by a single Dart object.
+///
+/// It is expected that when the C++ object is no longer needed [dispose] is
+/// called.
+///
+/// To prevent memory leaks, the underlying C++ object is deleted by the GC if
+/// it wasn't previously disposed of explicitly.
+class UniqueRef<T extends Object> {
+  UniqueRef(Object owner, T nativeObject, this._debugOwnerLabel) {
+    _nativeObject = nativeObject;
+    if (Instrumentation.enabled) {
+      Instrumentation.instance.incrementCounter('$_debugOwnerLabel Created');
+    }
+    _finalizationRegistry.register(owner, UniqueRef);
```

Comment:
should this be `finalizationRegistry.register(owner, this)`?

Reply:
Done. PTAL